deploy/Dockerfile

```diff
@@ -1,15 +1,17 @@
-# sha256 as of 2021-11-09
-FROM python:3.9-slim-bullseye@sha256:408de0cf1a057f5501ee6642ad24a4762738f63bacf09fb4c8d861669260b01e AS sphinx
+# sha256 as of 2023-01-17
+FROM python:3.9-slim-bullseye@sha256:8b0502fe3a8ae9d26567f66e217223595ff6abe901cba64fd8b1b58176311b27 AS sphinx
 
 ARG GIT_BRANCH=main
 RUN apt-get -q update && apt-get -qy upgrade && apt-get -qy install git make latexmk texlive-latex-extra
 COPY ./ .
-RUN pip install -r requirements/requirements.txt
+RUN pip install --upgrade pip
```

Comment:
I think you're right that we don't really need to if we keep up with Debian Stable. I've seen the `pip install --upgrade pip` strategy used in a few CI examples, and I think it may be motivated by wanting to avoid `pip` compatibility problems (see https://pythonspeed.com/articles/upgrade-pip/). But if we don't use very outdated distributions, and ensure we pull in the latest security upgrades into the container where we're using versioned images, we should be OK without that step. I've removed it.

Reply:
I think it didn't make sense in the way that we're currently pinning everything and yeah, if there is some compatibility issue, we should update the base container instead of pip alone.